test/impl-base/src/main/java/org/jboss/arquillian/test/impl/EventTestRunnerAdaptor.java

```diff
@@ -108,6 +127,15 @@ public TestResult test(TestMethodExecutor testMethodExecutor) throws Exception
       Validate.notNull(testMethodExecutor, "TestMethodExecutor must be specified");
 
       final List<TestResult> result = new ArrayList<TestResult>();
+      
+      TestExecutionDecider testExecutionDecider = resolveTestExecutionDecider(manager);
+
+      if (testExecutionDecider != null && !testExecutionDecider.execute(testMethodExecutor.getMethod()))
+      {
+          result.add(TestResult.skipped(null));
```

Comment:
In the JUnit integration we actually only check if testResult.hasException() {throw Exception}.. If the test is skipped the JUnit specific exception AssumptionViolatedException is caught from JUnit and rethrown.. We would need to update the JUnit integration as well to check for the State==skipped and recreate a AssumptionViolatedException